File: src/main/java/com/anod/appwatcher/content/TagsCursor.java
package com.anod.appwatcher.content;

import android.database.Cursor;

import com.anod.appwatcher.model.Tag;
import com.anod.appwatcher.model.schema.TagsTable;

/**
 * @author algavris
 * @date 10/03/2017.
 */

public class TagsCursor extends CursorWrapperCrossProcess {
    /**
     * Creates a cursor wrapper.
     *
     * @param cursor The underlying cursor to wrap.
     */
    public TagsCursor(Cursor cursor) {
        super(cursor);
    }

    public Tag getTag()
    {
        return new Tag(
                getInt(TagsTable.Projection._ID),
                getString(TagsTable.Projection.NAME),
                getInt(TagsTable.Projection.COLOR)
        );
    }
}

File: src/main/java/com/anod/appwatcher/content/TagsCursor.kt
package com.anod.appwatcher.content

import android.database.Cursor

import com.anod.appwatcher.database.entities.Tag
import com.anod.appwatcher.database.TagsTable
import info.anodsplace.framework.database.CursorIterator

/**
 * @author Alex Gavrishev
 * *
 * @date 10/03/2017.
 */

class TagsCursor(cursor: Cursor?) : CursorIterator<Tag>(cursor) {

    override val current: Tag
        get() = Tag(
                getInt(TagsTable.Projection._ID),
                getString(TagsTable.Projection.name),
                getInt(TagsTable.Projection.color)
        )
}
